File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.java
class Solution {
    public int maxProfit(int[] prices) {
        int left=0, right=1;
        int profit=0;
        while(right<prices.length){
            int diff=prices[right]-prices[left];
            profit=Math.max(diff,profit);
            if(prices[right]<prices[left]){
                left=right;
                right++;
            }else{
                right++;
            }
        }
        return profit;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


class Solution:
    def goodNodes(self, root: TreeNode) -> int:

        def dfs(node, max_sofar):
            if not node:
                return 0

            good_node = 0

            if node.val >= max_sofar:
                good_node += 1
            new_max = max(max_sofar, node.val)
            left = dfs(node.left, new_max)
            right = dfs(node.right, new_max)
            return left + right + good_node

        return dfs(root, root.val)
